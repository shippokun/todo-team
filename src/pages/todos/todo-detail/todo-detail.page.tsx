import * as React from "react";
import { useParams } from "react-router-dom";

import { TodoDetailContainer } from "./containers";

interface RouterParams {
  id: string;
}

const toId = (id: string): number => {
  return Number(id);
};

export const TodoDetailPage: React.FC = () => {
  const { id } = useParams<RouterParams>();

  return <TodoDetailContainer id={toId(id)} />;
};
