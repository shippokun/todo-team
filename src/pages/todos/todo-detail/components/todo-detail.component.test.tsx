import * as React from "react";
import { cleanup, render } from "@testing-library/react";

import { Todo } from "@prisma/client";

import { TodoDetail } from "./todo-detail.component";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

jest.mock("@/libs");
type Props = {
  todo: Todo;
  deleteFunc: (id: number) => Promise<void>;
};

describe("todo-detail.component.tsx", () => {
  afterEach(() => {
    cleanup();
  });

  it("render", () => {
    const history = createMemoryHistory();
    const props: Props = {
      todo: {
        id: 1,
        title: "test",
        completed: false,
        createdAt: new Date(),
        updatedAt: new Date(),
      },
      deleteFunc: jest.fn(),
    };

    const { asFragment } = render(
      <Router history={history}>
        <TodoDetail
          todo={props.todo}
          deleteFunc={() => props.deleteFunc(props.todo.id)}
        />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
