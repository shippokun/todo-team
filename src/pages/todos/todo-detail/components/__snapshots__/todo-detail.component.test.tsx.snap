// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo-detail.component.tsx render 1`] = `
<DocumentFragment>
  <table>
    <tbody>
      <tr>
        <td>
          id
        </td>
        <td>
          1
        </td>
      </tr>
      <tr>
        <td>
          タイトル
        </td>
        <td>
          test
        </td>
      </tr>
      <tr>
        <td>
          作成日
        </td>
        <td />
      </tr>
      <tr>
        <td>
          更新日
        </td>
        <td />
      </tr>
    </tbody>
  </table>
  <a
    href="/todos/edit/1"
  >
    編集する
  </a>
  <button>
    削除する
  </button>
</DocumentFragment>
`;
